File: src/test/java/org/jetbrains/pmdkotlin/PmdTest.java
package org.jetbrains.pmdkotlin;

import net.sourceforge.pmd.PMD;
import net.sourceforge.pmd.PMDConfiguration;
import org.jetbrains.pmdkotlin.lang.kotlin.KotlinLanguageModule;
import org.testng.annotations.BeforeTest;
import org.testng.annotations.Test;


public class PmdTest {

    @BeforeTest
    public void setUp() {}


    @Test
    public void emptyTest() {
        PMDConfiguration config = new PMDConfiguration();
        config.setRuleSets("/home/user/pmd-kotlin/src/main/resources/ruleset/kotlin/empty.xml");
        config.setReportFormat("html");
        config.setInputPaths("/home/user/pmd-kotlin/src/test/resources/org/jetbrains/pmdkotlin/emptyCodeTests/emptyTryCatchBlock.kt");
        config.setThreads(0);
        config.setDefaultLanguageVersion(new KotlinLanguageModule().getDefaultVersion());

        int x = PMD.doPMD(config);
        assert (x == 3);
    }

    @Test
    public void deprecatedTest() {
        PMDConfiguration config = new PMDConfiguration();
        config.setRuleSets("/home/user/pmd-kotlin/src/main/resources/ruleset/kotlin/deprecated.xml");
        config.setReportFormat("html");
        //config.setInputPaths("/home/user/pmd-kotlin/src/test/resources/org/jetbrains/pmdkotlin/deprecatedTest/TraitKeywordTest.kt");
        //config.setInputPaths("/home/user/pmd-kotlin/src/test/resources/org/jetbrains/pmdkotlin/deprecatedTest/EnumSuperConstructorTest.kt");
        //config.setInputPaths("/home/user/pmd-kotlin/src/test/resources/org/jetbrains/pmdkotlin/deprecatedTest/EnumDelimiterTest.kt");
        //config.setInputPaths("/home/user/pmd-kotlin/src/test/resources/org/jetbrains/pmdkotlin/deprecatedTest/LambdaSyntaxTest.kt");
        config.setInputPaths("/home/user/pmd-kotlin/src/test/resources/org/jetbrains/pmdkotlin/deprecatedTest/StaticAssertTest.kt");
        config.setThreads(0);
        config.setDefaultLanguageVersion(new KotlinLanguageModule().getDefaultVersion());
        int x = PMD.doPMD(config);
        assert (x == 1);
    }
}

File: src/test/kotlin/org/jetbrains/pmdkotlin/PmdTest.kt
package org.jetbrains.pmdkotlin

import net.sourceforge.pmd.PMD
import net.sourceforge.pmd.PMDConfiguration
import org.jetbrains.pmdkotlin.lang.kotlin.KotlinLanguageModule
import org.junit.Assert.assertEquals
import org.junit.Before
import org.junit.Test


public class PmdTest {

    @Before
    public fun setUp() {
    }

    @Test
    public fun emptyTest() {
        val config = PMDConfiguration()
        config.setRuleSets(getResourcePath("ruleset/empty.xml"))
        config.setReportFormat("html")

        config.setInputPaths(getResourcePath("emptyCodeTests/"))
        config.setDefaultLanguageVersion(KotlinLanguageModule().getDefaultVersion())

        assertEquals(3, PMD.doPMD(config))
    }

    @Test
    public fun deprecatedTest() {
        val config = PMDConfiguration()
        config.setRuleSets(getResourcePath("ruleset/deprecated.xml"))
        config.setReportFormat("html")
        //config.setReportFile("atata1.html")

        config.setInputPaths(getResourcePath("deprecatedTest/"))

        config.setDefaultLanguageVersion(KotlinLanguageModule().getDefaultVersion())

        assertEquals(10, PMD.doPMD(config))
    }

    @Test
    public fun otherTest() {
        val config = PMDConfiguration()
        config.setRuleSets(getResourcePath("ruleset/deadcode.xml"))
        config.setReportFormat("html")
        //config.setReportFile("atata.html")
        //config.setInputPaths(getResourcePath("deadCodeTests/UnusedFunctionTest.kt"))
        //config.setInputPaths(getResourcePath("deadCodeTests/UnusedVariableTest.kt"))
        config.setInputPaths(getResourcePath("deadCodeTests/UnreachableCodeTest.kt"))

        config.setDefaultLanguageVersion(KotlinLanguageModule().getDefaultVersion())
        assertEquals(5, PMD.doPMD(config))
    }
}
